### ISSUE TITLE
Word add-ins steal focus from document

### ISSUE THREAD

**Issue by fergiemcdowall** [OfficeDev/office-js#316]
After performing an action in an add-in, there seems to be no way to send focus back to the main document. The user is forced to manually click on the document.

(related: https://stackoverflow.com/questions/39888050/how-to-send-back-ms-word-cursor-focus-on-current-document-after-click-in-task-pa)

## Expected Behavior
I would expect there to be some way for the add-in to assign focus back to the document

## Current Behavior
Currently this works as expected on Word Online but doesn't work on latest versions of Word for Windows or Mac

## Steps to Reproduce

  1. Open a Word add-in in Word for Windows/Mac (not Word Online) and start typing
  2. Click on the add-in so that it has focus
  3. Continue typing in your document (nothing happens)
  4. Add-in has focus, and it can't be programatically returned to the document at the currently active cursor so the user has to manually click on the document to continue editing

## Context
I am a developer for [SmartLawyer Writer](https://appsource.microsoft.com/en-us/product/office/WA104381487?src=office&tab=Overview), a commercial lawtech add-in in the Norwegian add-in Store. Our users are legal professionals who write long documents. This problem means that our add-in (and other add-ins) can be seen as breaking a user's "flow" when they are writing.

## Your Environment
This bug seems to affect recent versions of Word on Windows and Mac, but works as intended on Word Online in all browsers.

**Comment by kbrandl** [comment]
@fergiemcdowall thanks for reaching out about this.

@JuaneloJuanelo are you able to provide input on this issue? (Please let me know if this should be directed to someone else.)

**Comment by JuaneloJuanelo** [comment]
as correctly answered on stack this is not supported today. @fergiemcdowall  if this is a pressing issue please open (or upvote)  a new request in our uservoice channel https://officespdev.uservoice.com/
. thanks.

**Comment by ThomasBarnekow** [comment]
I am experiencing the same issue today, three years later in December 2021. It really disturbs the flow!

**Comment by maximeburri** [comment]
Do you know if there is a plan to support that?

**Comment by lorenz-nlp** [comment]
Same here, will this issue never be approached?

**Comment by sasif2280** [comment]
is this issue is going to be adressed or not?

**Comment by pki89** [comment]
@fergiemcdowall @JuaneloJuanelo an update would be highly appreciated as this heavily disturbes UX.
ty

**Comment by jimbarrett33** [comment]
Same for Excel. This is very important for accessibility and use of keyboard only scenarios.

**Comment by marcelstrauss** [comment]
Hi, any updates on this?

**Comment by SPEuricom** [comment]
Its crazy that such a standard/logical usecase is ignored and just not possible in the current API....

**Comment by jimbarrett33** [comment]
Crickets...

**Comment by PsySolix** [comment]
I think this will never get fixed, OfficeJs is filled with caveats, inconsistencies or just weird behaviour..

**Comment by fergiemcdowall** [comment]
Hi @JuaneloJuanelo (or whoever is a current maintainer) - I'm not sure if its ok just to close bug reports or feature requests that get sustained support over time. Probably this issue, and others like it, should be parked in some sort of "future evaluation" queue rather than just "CLOSED."

**Comment by ErikSkaarup** [comment]
Is there any workaround for this to trigger focus? is there any of the commands available in API which "side-effect" is focus which can be hacked?

**Comment by DDJensMadsen** [comment]
Hi maintainers,

We’re currently in the process of porting our existing VSTO add-in to the Office.js platform, and the issue described here is proving to be a major blocker for us.

The inability to programmatically set or restore focus within the task pane severely limits usability. Users are forced to manually click to regain focus—there’s not even a keyboard shortcut available. This creates a frustrating user experience and disrupts key workflows.

This limitation is, unfortunately, a showstopper for our migration. Without a resolution or viable workaround, we may be forced to stick with our legacy VSTO implementation, despite its deprecation. In the VSTO API, the `Word.Document` has an array of `Window` objects. The `Window` object has a `SetFocus` method which allows us to control focus. 

Since this issue has been open for quite some time, and it appears that some of the originally involved Microsoft team members may no longer be actively maintaining this area, we hope it’s acceptable that we tag a few current maintainers (@RuizhiSunMS @jeremy-msft @exextoc) directly. Our intention is simply to bring renewed attention to what we believe is a very important and time-sensitive issue.

We’re more than happy to contribute in any way—whether through testing, feedback, or code—if it helps move this forward. Thank you again for your continued work on Office.js, and we hope this issue can be prioritised.

**Comment by RuizhiSunMS** [comment]
Hi @DDJensMadsen , @ErikSkaarup @fergiemcdowall @ErikSkaarup @jimbarrett33 @marcelstrauss @SPEuricom 
I did a quick look at this. May I confirm with you that you need a new feature or a workaround as least to move focus back?
@AmandaHan123 , please help triage.

**Comment by DDJensMadsen** [comment]
@RuizhiSunMS Thank you so much for the quick and positive response—it's truly appreciated!

We’ve already reviewed the Word API reference and the Common API documentation in detail, but unfortunately haven’t been able to identify any viable workaround so far. If there is one, we'd be very grateful to learn about it.

In the meantime, a temporary workaround to programmatically restore focus would be very helpful and would unblock us in the short term. Of course, having a proper feature—something along the lines of window.setFocus()—would be ideal in the long run.

Thanks again, and we’re happy to support or test any proposed solutions.

**Comment by AmandaHan123** [comment]
Hello @DDJensMadsen @ErikSkaarup @fergiemcdowall @PsySolix @jimbarrett33 @SPEuricom @marcelstrauss @pki89 @sasif2280 @lorenz-nlp @maximeburri @ThomasBarnekow @JuaneloJuanelo,

Thank you bringing this to our attention! Unfortunately we do not have a plan on this yet and there is no workaround to programmatically restore focus. We recommend you to upvote or submit a new Tech Community New Ideas item, this will give us visibility of the commonality of it and will help with our prioritization. [Microsoft 365 Developer Platform - Microsoft Community Hub](https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/idb-p/Microsoft365DeveloperPlatform)

**Comment by DDJensMadsen** [comment]
Thank you for the follow-up @AmandaHan123  .

We understand that there’s currently no workaround or plan in place, but we’d like to respectfully ask you to reconsider this request. Restoring focus programmatically is a very basic and essential capability for any interactive UI experience, especially in professional Office add-ins. From our perspective, this isn’t just a “nice-to-have” feature, but rather foundational functionality that users naturally expect to behave consistently.

We’re aware of at least two existing feature requests in the Tech Community Hub related to this issue. In our experience, submitting requests there rarely leads to concrete outcomes or clear feedback. That’s why we felt compelled to raise this directly here, to stress that this gap is actively blocking real-world migration efforts.

Without a roadmap, workaround, or commitment, our only viable option may be to continue relying on our VSTO add-in, despite the platform being deprecated. We would much prefer to invest in the Office.js platform and modernize our solution, but this limitation remains a significant barrier.

We would greatly appreciate it if the team (@RuizhiSunMS ) could at least clarify whether this functionality is being considered for the roadmap, or if there's any timeframe in which a decision might be made.

Thanks again for your time and engagement.

**Comment by njlgmeadow** [comment]
For anyone that are facing this issue i recommend upvoting this feature request, in hopes of providing some visibility to the team.
[set selection and focus back to document after action in ms-word web addin pane](https://techcommunity.microsoft.com/idea/microsoft365developerplatform/set-selection-and-focus-back-to-document-after-action-in-ms-word-web-addin-pane/4065642)

**Comment by AmandaHan123** [comment]
Hi @DDJensMadsen , thank you for sharing your thoughts. Personally I agree wih you this is a solid request to being able to set focus programmatically. The Word JS API team is using Tech Community Hub as a platform to gather all the feature requests from customers and the votes. During regularly planning stage, the team will consider them together with other backlogs. Please do upvote [set selection and focus back to document after action in ms-word web addin pane](https://techcommunity.microsoft.com/idea/microsoft365developerplatform/set-selection-and-focus-back-to-document-after-action-in-ms-word-web-addin-pane/4065642) and add your comments to the item. You may also consider to add a link to this GitHub issue. The vote count and the item count for the request on Tech Community Hub can help the team to set a proper priority during planning. Thank you for your understanding and your contribution!

**Comment by DDJensMadsen** [comment]
@AmandaHan123 Thank you again for the thoughtful response and for confirming that this is considered a solid request.

We’ve already followed your suggestion and linked this GitHub issue to the relevant item in the Tech Community Hub, where we've also added our upvote and comments.

That said, from a product development perspective, it would be helpful to have some visibility into your planning cadence. Specifically: how often are feature requests reviewed and prioritised, and what kind of timeline is typically expected between consideration and potential implementation? We're currently at a critical decision point—whether to proceed with rolling out our Office.js-based add-in or to remain with our VSTO implementation. We’ve already made a significant investment in the new Office.js add-in and would strongly prefer to move forward with it, but a few key limitations are acting as blockers.

In addition to the inability to programmatically restore focus, we’re also impacted by two other issues:

The [table selection issue](https://github.com/OfficeDev/office-js/issues/5388), where the Word API only returns the last row in a user selection within a table, making it impossible to reliably detect multi-row selections.

The lack of control over context menus (right-click behaviour), limits several important interaction patterns in our UX.

All three of these capabilities are available in our current VSTO add-in, so their absence in Office.js represents real regressions in functionality that are difficult to work around.

While we fully respect your internal planning processes, we also wanted to share, purely from a user perspective, that having both GitHub and the Tech Community Hub as channels for submitting bugs and feature requests can be confusing. GitHub tends to feel more transparent and directly connected to the engineering team, which is why we—and likely many others—raise issues here when facing blocking functionality gaps.

We truly appreciate the engagement so far and would be very grateful for any additional insight into planning timelines, prioritisation cycles, or whether this functionality is under active consideration for an upcoming release.

**Comment by AmandaHan123** [comment]
Hi @DDJensMadsen , thank you for sharing the information. I understand your condition now. The feature request on Tech Community has higher priority when team review all requests and backlogs during planning. However, considering you are at a critical decision point now, I have escalated your requirement (#10390864) to the Word JS API team to draw their attentions before next planning starts. Hope it can help you. I will update you here if there is any response.

**Comment by DDJensMadsen** [comment]
Thank you very much, @AmandaHan123 — we truly appreciate the escalation.

Just to summarise, the three key issues we’re facing are:

[Table selection](https://github.com/OfficeDev/office-js/issues/5388)

[Focus control](https://github.com/OfficeDev/office-js/issues/316). This issue :-) 

[Lack of control over context menus](https://github.com/OfficeDev/office-js/issues/5957) – we've just created this new issue to track it specifically.

Thanks again for your support. We're very grateful for your help in drawing attention to these blockers.

**Comment by kbrilla** [comment]
For those following this thread — we've compiled a community open letter to Microsoft 
based on a systematic analysis of all 6,200+ issues in this repository. It documents 
patterns of silent breaking changes, auto-closed bugs, platform fragmentation, and the 
financial/reputation impact on ISVs and developers.

If any of this resonates with your experience, your voice would strengthen the message: 
[[link to open letter issue]](https://github.com/OfficeDev/office-js/issues/6513)

This is entirely voluntary — we just wanted to make sure affected developers are aware.

**Comment by fergiemcdowall** [comment]
> If any of this resonates with your experience, your voice would strengthen the message:
> https://github.com/OfficeDev/office-js/issues/6513

@kbrilla interesting to see that the open letter names this thread as the oldest unresolved bug on the repo!
